examples/connext_dds/compression/README.md

```diff
@@ -0,0 +1,67 @@
+# Example Code: Data Compression
+
+## Concept
+
+The compression feature allows compressing the *USER_DATA* samples sent by a
+*DataWriter*.
+Compression can be configured by 3 QoSs (Check the ConnextDDS_CoreLibraries
+user-manual for more information):
+
+- compression_ids.
+- writer_compression_threshold.
+- writer_compression_level
+
+## Example description
+
+This example shows how to use compression. You will learn how to:
+
+-   Configure the compression feature by Qos
+-   How the Bandwidth saving is affected by each compression algorithm.
+Configurable by the command line parameter `--compression-id`.
+
+This test measures the number of bytes received at UDP transport-level on
+the subscriber side, this way we can compare how much bandwidth has been saved
+by enabling/disabling compression and the different compression algorithm.
+
+The *USER_DATA* can be also configured by `--input-file` command line parameter,
+where each of the lines on the files will represent a sample. This is a simple
+way to test and compare different kinds of data.
+
+## Publisher
+
+The publisher will create a list of samples from the input file (or 1K samples
+fill with zeros if no file is provided). Then will send all the samples on a
+loop until hits the sample count (--sample-count). The selected compression
+(--compression-id) will apply to each of the samples.
+
+A generic text file has been added along with this examples, you can find it
+under `resource/messages/text_payload.txt`.
+
+
+Execute the publisher with `--help` command line parameter for more information.
+
+Example of a publisher execution:
+
+- `compression_publisher -c ZLIB -i ../../../../../resources/messages/text_payload.txt -s 100`
```

Comment:
To avoid the ["long line" error](https://github.com/rticommunity/rticonnextdds-examples/runs/2021531829#step:5:5) from the linter you can use the code block Markdown feature

```bash
compression_publisher -c ZLIB -i ../../../../../resources/messages/text_payload.txt -s 100
```